### DISCUSSION TITLE
The assistant stops the conversation and throws an error.

### DISCUSSION THREAD

**Discussion by wupengfei0904** [danny-avila/LibreChat#2796] [General]
When using the assistant, if I stop the conversation while running the code interpreter and then send a new message to the assistant, it throws an error.

{"code":null,"error":{"code":null,"message":"Can't add messages to thread_lHOkfjbDzSYK44KCL6qwC0HB while a run run_r94CF1FEVmqjtCbFjARaEg32 is active.","param":null,"type":"invalid_request_error"},"headers":{"alt-svc":"h3=\":443\"; ma=86400","cf-cache-status":"DYNAMIC","cf-ray":"886883d5c8b17ec6-LAX","connection":"keep-alive","content-length":"217","content-type":"application/json","date":"Mon, 20 May 2024 01:22:19 GMT","openai-organization":"user-2jgtojzsiwlhjiltfamvjsoa","openai-processing-ms":"28","openai-version":"2020-10-01","server":"cloudflare","set-cookie":"__cf_bm=dyUxMIAbCgJV3tj4PJyM513xVrzqcwiNpd0YiNnTOfM-1716168139-1.0.1.1-zzkq0WFViVBzj12WRxkkocsrUQtcDhAhJ7NpvECOVLnk4u.fcqeMBphRt5rqnXWSagUurKVaiElOc0FMayuKiQ; path=/; expires=Mon, 20-May-24 01:52:19 GMT; domain=.api.openai.com; HttpOnly; Secure; SameSite=None, _cfuvid=OI9EflHc8Wel.GRPDnoejBN5St9dlC7ggrZGNjgi4wQ-1716168139241-0.0.1.1-604800000; path=/; domain=.api.openai.com; HttpOnly; Secure; SameSite=None","strict-transport-security":"max-age=15724800; includeSubDomains","x-request-id":"req_0a12e38eb46d453f33c4298f0173e898"},"level":"error","message":"[/assistants/chat/] 400 Can't add messages to thread_lHOkfjbDzSYK44KCL6qwC0HB while a run run_r94CF1FEVmqjtCbFjARaEg32 is active.","param":null,"request_id":"req_0a12e38eb46d453f33c4298f0173e898","stack":"Error: 400 Can't add messages to thread_lHOkfjbDzSYK44KCL6qwC0HB while a run run_r94CF1FEVmqjtCbFjARaEg32 is active.\n    at APIError.generate (/app/api/node_modules/openai/error.js:45:20)\n    at OpenAI.makeStatusError (/app/api/node_modules/openai/core.js:263:33)\n    at OpenAI.makeRequest (/app/api/node_modules/openai/core.js:306:30)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async initThread (/app/api/server/services/Threads/manage.js:31:21)\n    at async initializeThread (/app/api/server/controllers/assistants/chatV2.js:404:22)\n    at async Promise.all (index 0)\n    at async chatV2 (/app/api/server/controllers/assistants/chatV2.js:451:5)","status":400,"type":"invalid_request_error"}

**Comment by danny-avila** [comment]
I'm looking into this

**Comment by danny-avila** [comment]
Can't reproduce, what is the latest commit you are on?

![image](https://github.com/danny-avila/LibreChat/assets/110412045/54c1fe9c-f996-4c2b-9225-6e35b705c67d)

**Reply by wupengfei0904** [reply]
I often encounter this issue when the assistant is running. For example, during the output of some longer content, I click the 'Pause Output' button and then send a new request, but it results in an error. It seems that the assistant is still running in the background, meaning that the 'Pause Output' button did not timely stop the assistant's operation. You can try with a longer content output, then pause the conversation and send a new request.

**Reply by wupengfei0904** [reply]
I want to express that there might be an issue with pausing the process from the front end while the assistant process is running. Is that the case?

**Reply by danny-avila** [reply]
I understand, that is what I did and I could not reproduce your issue.

There could be a timing issue indeed, are you able to reliably reproduce it? Can you share a video?
